backend/src/server/plugins/add-errors-to-response-schemas.ts

```diff
@@ -3,9 +3,12 @@ import fp from "fastify-plugin";
 
 import { DefaultResponseErrorsSchema } from "../routes/sanitizedSchemas";
 
+const isScimRoutes = (pathname: string) =>
+  pathname.startsWith("/api/v1/scim/Users") || pathname.startsWith("/api/v1/scim/Groups");
+
 export const addErrorsToResponseSchemas = fp(async (server) => {
```

Comment:
It's just for SCIM because scim has a very different format for error messages. For others this is not an issue. Unless the schema changes

Reply:
But we can't guarantee that other auth providers will only return error  codes that we have defined. Our error schema is too generic to be useful and it might make it more difficult to find other issues. If the error codes are anything else, it will not get shown:

```
export const DefaultResponseErrorsSchema = {
  400: z.object({
    statusCode: z.literal(400),
    message: z.string(),
    error: z.string()
  }),
  404: z.object({
    statusCode: z.literal(404),
    message: z.string(),
    error: z.string()
  }),
  401: z.object({
    statusCode: z.literal(401),
    message: z.any(),
    error: z.string()
  }),
  403: z.object({
    statusCode: z.literal(403),
    message: z.string(),
    error: z.string()
  }),
  500: z.object({
    statusCode: z.literal(500),
    message: z.string(),
    error: z.string()
  })
};
``` 

This should be case by case basis for end points. We should remove this handler completely

Reply:
Not really this is a scim specific issue because everywhere else we follow Infisical api format. Expect scim which expects different.

Also generics make sense technically errors should be only shown in controlled format. Showing too much information is kinda bad too from a security perspective.

Reply:
https://docs.hidglobal.com/activid-appliance-v8.6/api/scim/scim-error-handling.htm

Reply:
First yes its logged and the response will be stating that its a response error.

Now coming back to the main issue. All other errors that we don't handle are fallback to internal server error. And that will be clearly stated in response and log.